src/CG/CG_elem.cpp

```diff
@@ -2113,6 +2113,829 @@ bool intersectSegmentPolygon( array3D const &P0, array3D const &P1, std::size_t
     return false;
 }
 
+/*!
+ * Check if a pixel is crossing a given plane and fully calculate the intersection (a segment) with 
+ * edges, eventually. A pixel is defined as a planar rectangular, vertical to a cartesian axis.
+ * The polygon greated from the intersection belongs to the subspace opposite of the
+ * one pointed by the plane's normal
+ *
+ * Given the normal cartesian direction dir, the pixel coordinates
+ * are represened on its 2D plane as (x, y) where x = (dim + 1) % 3 and y = (dim + 2) % 3 
+ *
+ * Local indeces for edges are defined as follows:
+ * edge 0: y = y min
+ * edge 1: x = x max
+ * edge 2: y = y max
+ * edge 3: x = x min
+ *
+ * Co-planarity of input pixel and plane will produce no intersection.
+ * No intersection is considered if the plane passes from a pixel's vertex,
+ * and no other intersection exist.
+ *
+ * \param[in] Pp point belonging to the plane
+ * \param[in] nP unit normal to plane
+ * \param[in] A0 min point of pixel on the 2D space defined by the pixel direction dir
+ * \param[in] A1 max point of pixel on the 2D space defined by the pixel direction dir
+ * \param[in] dir is an integer from 1 to 3 and indicates the cartesian direction normal
+ * to the pixel. Its sign affects the polynomial's orientation. The above used dim variable
+ * is computed as dim = |dir| - 1
```

Comment:
What's the point of this argument? You have the coordinates of the vertices of the pixel ordered in a well defined way, this should be enough.

Reply:
It was needed, because the vertices weren't given as arguments to the functions. This has changed. By giving the vertices coordinates, most of those arguments are not needed any more.